File: trunk/theLink/example/perl/manfilter.pl
#+
#§  \file       theLink/example/perl/manfilter.pl
#§  \brief      \$Id$
#§  
#§  (C) 2009 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev$
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§

use strict;
use Net::PerlMsgque;

sub FTRcmd {
  my $ctx = shift;
  my $ftr = $ctx->ServiceGetFilter();
  $ftr->SendSTART();
  while ($ctx->ReadItemExists()) {
    $ftr->SendC("<" . $ctx->ReadC() . ">");
  }
  $ftr->SendEND_AND_WAIT("+FTR");
  $ctx->SendRETURN();
}

package main;

  our $srv = new Net::PerlMsgque::MqS();
  eval {
    $srv->ConfigSetName("ManFilter");
    $srv->ConfigSetIsServer(1);
    $srv->ConfigSetFactory(sub {new Net::PerlMsgque::MqS()});
    $srv->LinkCreate(@ARGV);
    $srv->ServiceCreate("+FTR", \&FTRcmd);
    $srv->ServiceProxy("+EOF");
    $srv->ProcessEvent({wait => "FOREVER"});
  };
  if ($@) {
    $srv->ErrorSet($@);
  }
  $srv->Exit();

File: trunk/theLink/example/go/manfilter.go
/**
 *  \file       theLink/example/go/manfilter.go
 *  \brief      \$Id$
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev$
 *  \author     EMail: aotto1968 at users.berlios.de
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */

package main

import (
  . "gomsgque"
    "os"
)

type ManFilter MqS
  func (this *ManFilter) Factory () *MqS {
    return NewMqS(nil)
  }

type FTR MqS
  func (this *FTR) Call () {
    ctx := (*MqS)(this)
    ftr := ctx.ServiceGetFilter()
    ftr.SendSTART()
    for ctx.ReadItemExists() {
      ftr.SendC("<" + ctx.ReadC() + ">")
    }
    ftr.SendEND_AND_WAIT2("+FTR")
    ctx.SendRETURN()
  }

func main() {
  var srv = NewMqS(nil)
  defer func() {
    if x := recover(); x != nil {
      srv.ErrorSet(x)
    }
    srv.Exit()
  }()
  srv.ConfigSetName("ManFilter")
  srv.ConfigSetIsServer(true)
  srv.ConfigSetFactory((*ManFilter)(srv))
  srv.LinkCreate(os.Args...)
  srv.ServiceCreate("+FTR", (*FTR)(srv))
  srv.ServiceProxy("+EOF")
  srv.ProcessEvent(WAIT_FOREVER)
}

